delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -1032,6 +1035,40 @@ def do_rest_call(self, url, data):
             LOG.error(err_msg)
             raise exception.InvalidResults(err_msg)
 
+    @staticmethod
+    def future_callback(future, res_list):
+        if future and future.result():
+            res_list.append(future.result())
+
+    def performance_task(self, **kwargs):
+        json_info = self.do_rest_call(kwargs.get('rest_url'), None)
+        metrics = PerformanceHandler.get_perf_value(
+            metrics=kwargs.get('metrics'),
+            storage_id=kwargs.get('storage_id'),
+            start_time=kwargs.get('start_time'),
+            end_time=kwargs.get('end_time'),
+            data_info=json_info,
+            resource_id=kwargs.get('resource_id'),
+            resource_name=kwargs.get('resource_name'),
+            resource_type=kwargs.get('resource_type')
+        )
+        return metrics
+
+    def do_method_in_pool(self, kwargs_list):
+        res_list = []
+        with ThreadPoolExecutor(max_workers=4) as pool:
+            future_list = []
+            for kwargs in kwargs_list:
+                future = pool.submit(self.performance_task, **kwargs)
+                future_list.append(future)
+                future.add_done_callback(
+                    functools.partial(
+                        NetAppHandler.future_callback,
+                        res_list=res_list))
+            as_completed(future_list)
+            pool.shutdown()
```

Comment:
If you use 'with', it will close the thread pool by itself, no need to use the shutdown method.